.bumpversion.cfg

```diff
@@ -0,0 +1,16 @@
+# https://github.com/c4urself/bump2version/issues/42
```

Comment:
let's chat about this.  my preferred version maintenance system at this point is https://pypi.org/project/setuptools-scm/ ... after perhaps a little time to understand how it works, it's a fantastic way to single-source your version info, while also encouraging good git tagging practice

Reply:
I'm happy to switch over! Once swapped over, does this mean that the only command to run to "bump version" will be `git tag vX.Y.Z`? I guess the only knowledge I care about is "what is the maintainer usage to bump version".

Reply:
yeah, the way it works is that versions are sourced entirely from git tags.  So:
- you label a specific _commit_ as a version with an [_annotated git tag_](https://git-scm.com/book/en/v2/Git-Basics-Tagging) (not a lightweight tag).
    - `git tag -a vX.Y.Z -m X.Y.Z`
- when you create the sdist or wheel, setuptools_scm acts as a plugin to setuptools that injects the version into the package metadata at build time (so, setuptools needs to be in the build environment...)
   - `setuptools-scm` must be in the `build.requires` section of `pyproject.toml`
   - `setuptools_scm` must go in the `host` section of a conda-forge recipe
- that's pretty much it, but to add a `__version__` attribute to your main module, you also do one of [these two things](https://github.com/pypa/setuptools_scm/#retrieving-package-version-at-runtime)  (i prefer the first)

Reply:
Ahhhh cool. Feel free to swap over to the setuptools_scm then! Thanks for the info!

Reply:
ok, will do!  the first time or two might be a little different or feel a little mysterious.  but it's treated me well, and it seems to gathering momentum as "the" way to do it (at least for setuptools).   i promise to be on call to help out if need be for the next couple releases :)